image.go

```diff
@@ -0,0 +1,102 @@
+package dalec
+
+import (
+	"github.com/google/shlex"
+	"github.com/moby/buildkit/exporter/containerimage/image"
+	"github.com/pkg/errors"
+)
+
+// ImageConfig is the configuration for the output image.
+// When the target output is a container image, this is used to configure the image.
+type ImageConfig struct {
+	// Entrypoint sets the image's "entrypoint" field.
+	// This is used to control the default command to run when the image is run.
+	Entrypoint string `yaml:"entrypoint,omitempty" json:"entrypoint,omitempty"`
+	// Cmd sets the image's "cmd" field.
+	// When entrypoint is set, this is used as the default arguments to the entrypoint.
+	// When entrypoint is not set, this is used as the default command to run.
+	Cmd string `yaml:"cmd,omitempty" json:"cmd,omitempty"`
+	// Env is the list of environment variables to set in the image.
+	Env []string `yaml:"env,omitempty" json:"env,omitempty"`
+	// Labels is the list of labels to set in the image metadata.
+	Labels map[string]string `yaml:"labels,omitempty" json:"labels,omitempty"`
+	// Volumes is the list of volumes for the image.
+	// Volumes instruct the runtime to bypass the any copy-on-write filesystems and mount the volume directly to the container.
+	Volumes map[string]struct{} `yaml:"volumes,omitempty" json:"volumes,omitempty"`
+	// WorkingDir is the working directory to set in the image.
+	// This sets the directory the container will start in.
+	WorkingDir string `yaml:"working_dir,omitempty" json:"working_dir,omitempty"`
+	// StopSignal is the signal to send to the container to stop it.
+	// This is used to stop the container gracefully.
+	StopSignal string `yaml:"stop_signal,omitempty" json:"stop_signal,omitempty" jsonschema:"example=SIGTERM"`
+	// Base is the base image to use for the output image.
+	// This only affects the output image, not the intermediate build image.
+	Base string `yaml:"base,omitempty" json:"base,omitempty"`
+
+	// Post is the post install configuration for the image.
+	// This allows making additional modifications to the container rootfs after the package(s) are installed.
+	//
+	// Use this to perform actions that would otherwise require additional tooling inside the container that is not relavent to
+	// the resulting container and makes a post-install script as part of the package unnecessary.
+	Post *PostInstall `yaml:"post,omitempty" json:"post,omitempty"`
+
+	// User is the that the image should run as.
+	User string `yaml:"user,omitempty" json:"user,omitempty"`
+}
+
+// MergeImageConfig copies the fields from the source [ImageConfig] into the destination [image.Image].
+// If a field is not set in the source, it is not modified in the destination.
+// Envs from [ImageConfig] are merged into the destination [image.Image] and take precedence.
+func MergeImageConfig(dst *image.Image, src *ImageConfig) error {
```

Comment:
Can `dst` actually be a pointer to the config?

Reply:
Done.